### ISSUE TITLE
Porting to Plasma 6

### ISSUE THREAD

**Issue by ciachciarachciach666** [ismailof/mediacontroller_plus#32]
Hey ismailof,

Since this is the best media controller addon, are there any chances of it getting ported to Plasma 6?

Thanks!

**Comment by ismailof** [comment]
I'll start with a bit of digression:

For media players in particular there are two points that require porting: 
- The general plasmoid API changes for Plasma 6: this is already done 
- The deprecation of the previous Media interface (mpris) from using DataEngines, which got deprecated and replaced by a new plugin. 

Were only for the first part, I would have publish the current porting work. But the second point is a bit more tedious, and probably the easiest way is to copy the changes from the official plasmoid. 

At this point the official applet already offers two of the characteristics why I started this "fork": panel representation with the song info, and icon tabbar for different players. So I'm wondering if the effort would be best invested into adding the last remaining bits to the official applet (options for formatting the panel view, and panel controls) instead of the inverse.

So, TLDR, I cannot promise there will be a ported version soon :) but I'm considering it. 

Thanks for your interest!

**Comment by ciachciarachciach666** [comment]
Thanks a lot for the detailed answer! TBH for me the most important parts are control buttons on the taskbar. I've checked and there are no plasmoids with that functionality yet.

**Comment by MurzNN** [comment]
The same for me, control buttons on the panel is the thing!

**Comment by metzger100** [comment]
I don't know if it helps, but I guess that most of the users of "mediacontroller plus" are using it because of the panel controls. The official applet is very limited (There isn't even a settings page). I like the official applet but the missing panel controls are a game breaker for me. On the other side I always had problems with "mediacontroller plus" while using brave browser and the spotify flatpak (other flatpaks had the same problem). The automatic player tab never got it right and selected the paused youtube video in the browser instead of the running spotify song. The official applet is working flawlessly in this case. In my humble opinion an upgrade of the official applet would be better than an update of mediacontroller plus.

**Comment by jediafr** [comment]
@ismailof 
I reverted back to plasma 5 for your widget (and alias). Your widget takes the all menu available space, let me adjust the volume by scrolling in there and have the controls icons also, it is very readable and i keep track of the song progression visually. Please consider porting it , it is that awesome ! 
Some plasma 6 have some of that but not all

**Comment by Exanima98** [comment]
Loved it on plasma 5, it's really a shame not to see it on plasma 6. The buttons on the media player are super useful

**Comment by ReillyBrogan** [comment]
I think that adding the improvements to the upstream widget is the "correct" approach as it will bring the improvements to the most users. However, it also means that the changes wouldn't be available until Plasma 6.1 at minimum which is not ideal. Perhaps use this project as a "staging" area to get those ready for upstreaming before opening MRs to add the improvements to the original widget?

**Comment by Sadi58** [comment]
Finally, the wait is over, it seems: **MMedia Player**
https://store.kde.org/p/2342323
https://github.com/MCC45TR/kde-plasma6-widgets/tree/main/music-player
Although it has some drawbacks...
